docs/typescript/reference/dbos.md

```diff
@@ -0,0 +1,953 @@
+---
+
+# DBOS Class
+
+The `DBOS` class is the heart of the DBOS Transact programming API, providing decorators to identify key application functions, and accessors for context-dependent state.
+
+Note that this is the second major version of the API.  The first version of the API was based on passing [contexts](./contexts.md) and used a separate set of [decorators](./decorators.md).  It is fine to use both approaches in the same DBOS application, however the `DBOS` class is simpler and is therefore recommended.
+
+## Decorating Workflows, Transactions, and Steps
+DBOS provides reliability guarantees for programs that are [written as workflows comprised of steps, transactions, and child workflows](../programming-guide.md).  The workflow, transaction, and step functions are marked with decorators.
+
+### DBOS.workflow
+`@DBOS.workflow` registers a function as a DBOS workflow.  
+
+```typescript
+@DBOS.workflow(config?: WorkflowConfig)
+```
+
+When called, `workflow` functions are wrapped such that they will be [run to completion exactly once](../tutorials/idempotency-tutorial.md).
+
+#### Decorating Static Methods
+Workflow functions must be class members, and may be `static`.  To mark a `static` member function as a workflow, simply place the `@DBOS.workflow()` decorator above it.
+
+Example:
+```typescript
+import { DBOS, ConfiguredInstance, WorkflowQueue } from '@dbos-inc/dbos-sdk';
+
+class Workflows
+{
+  // Mark a static method as a DBOS workflow
+  @DBOS.workflow()
+  static async processWorkflow(value: string) {
+    ...
+  }
+}
+```
+
+#### Decorating Instance Methods
+Workflow functions may also be instance methods.
+
+This requires a few more steps.  For instance methods to work, each class instance must be given a name and registered with DBOS, so that a corresponding instance can be found for workflow recovery.  This is done with `DBOS.configureInstance`:
+
+```typescript
+DBOS.configureInstance(cls: Constructor, name: string, ...args: unknown[]) : R
+```
+
+The arguments to `DBOS.configureInstance` are:
+* `cls`: The class to be instantiated and configured.  This class must extend `ConfiguredInstance`.
+* `name`: The name for the configured instance (which must be unique within the set of instances of the same class)
+* `...args`: Any additional arguments to pass to the constructor function `cls`.
+
+Configured classes must extend from the following abstract class:
+```typescript
+export abstract class ConfiguredInstance {
+  readonly name: string;
+  constructor(name: string) {
+    this.name = name;
+  }
+  abstract initialize(ctx: InitContext): Promise<void>;
+}
+```
+
+The `ConfiguredInstance` base class provides naming information, and also declares the `initialize` function which will be called after the DBOS runtime has started, but before workflow processing commences.  The argument to `initialize` is an [`InitContext`](../reference/contexts.md#initcontext), which provides access to configuration and other information.
```

Comment:
Since we're getting away from contexts, do we want to remove the need of `InitContext` here as well? We can directly obtain connections through the `DBOS` class.

Reply:
We can, that would involve a change to initialize in the codebase before we change the doc.

Reply:
It's on my list, was just lower priority, I got convinced people are not using init much.

Reply:
Sounds good.